docs/rfcs/rfc-019-optional-features.md

```diff
@@ -0,0 +1,68 @@
+# Optional features
+## Problem
+A Neo4j GraphQL library design introduced is to add operators directly in the schema for the supported fields.
+The number of operators and other features grows and with them the size of the `augmented schema`.
+A solution to reduce the size of the problem is to introduce the possibility to configure the environment with only the features necessary.
+
+## Proposed Solution (Config)
+The Config solution is to extend the `Neo4jGraphQLConfig` to support the list of features to be included with the `features` attribute.
+
+### Usage Examples
+```javascript
+  const neoSchema = new Neo4jGraphQL({
+        config: {
+            enableDebug: true,
+            features: {
+              stringComparator: { enabled: true, settings: {...} }
+            }
```

Comment:
I was personally thinking something more along the lines of:

```suggestion
            features: {
              filters: {
                String: {
                  GT: true,
                  LT: true,
                }
              }
            }
```

The would make it significantly more extensible, and more expressive in my opinion, for example:

```
            features: {
              filters: {
                String: {
                  GT: true,
                  LT: true,
                  CONTAINS: false
                }
                Point: {
                  DISTANCE: false
                }
              }
            }
```

Reply:
Thank you for the suggestion! I really like the idea to have the common interface `filters` in the settings, so that could be used also retroactively with the current operators. I will not include that level of details like `GT: true`, as then the implementation will have to add a check for any comparators used, rather than using a more generic check. In my opinion, the cases where a user needs `GT` but not `LT` are rare, and even more considering that in a case when they discover later that they need that, they'll have to regenerate the schema without being able to modify it at runtime. 
What about something like this:
```javascript
features: {
  filters: {
    String: {
      COMPARATORS: true,
      STRING_FUNCTIONS: false
    }
    Point: {
      DISTANCE: false
    }
  }
}
```
Thoughts?

Reply:
The granularity is a good question. I agree with Simone, it seems unlikely that a user wants to allow `LT` but not for instance `LTE` or `GT`. The middle ground would be for instance a `COMPARATORS_LT` and `COMPARATORS_GT` setting but that probably does not add too much.

Reply:
While I agree on those points, it is important to take into account that a user may not intuitively know what a "Comparator" or a "String_function" is

Using the same names as the query itself makes it obvious. Do you want to use GT? enable the GT filter.

Reply:
Let's move forward with the @darrellwarde suggestion so!